tutorials/scripting/debug/custom_performance_monitors.rst

```diff
@@ -0,0 +1,117 @@
+  instance, in a voxel game, you could create a performance monitor to track the
+  number of chunks that are loaded every second.
+- Displaying in-game metrics that are not strictly related to performance, but
+  are still useful to graph for debugging purposes. For instance, you could
+  track the number of enemies present in the game to make sure your spawning
+  mechanic works as intended.
+
+Creating a custom performance monitor
+-------------------------------------
+
+In this example, we'll create a custom performance monitor to track how many
+enemies are present in the currently running project.
+
+The main scene features a :ref:`class_Timer` node with the following script attached:
+
+::
+
+    extends Timer
+
+
+    func _ready():
+        # The slash delimiter is used to determine the category of the monitor.
+        # If there is no slash in the monitor name, a generic "Custom" category
+        # will be used instead.
+        Performance.add_custom_monitor("game/enemies", get_enemy_count)
+        timeout.connect(_on_timeout)
+        # Spawn 20 enemies per second.
+        wait_time = 0.05
+        start()
+
+
+    func _on_timeout():
+        var enemy = preload("res://enemy.tscn").instantiate()
+        get_parent().add_child(enemy)
+
+
+    # This function is called every time the performance monitor is queried
+    # (this occurs once per second in the editor, more if called manually).
+    # The function must return a number greater than or equal to 0 (int or float).
+    func get_enemy_count():
+        return get_tree().get_nodes_in_group("enemies").size()
+
+
+``enemy.tscn`` is a scene with a Node2D root node and Timer child node. The
+Node2D has the following script attached:
+
+::
+
+    extends Node2D
+
+
+    func _ready():
+        add_to_group("enemies")
+        $Timer.timeout.connect(_on_timer_timeout)
+        # Despawn enemies 2.5 seconds after they spawn.
+        $Timer.wait_time = 2.5
+        $Timer.start()
+
+
+    func _on_timer_timeout():
+        queue_free()
+
+In this example, since we spawn 20 enemies per second, and each enemy despawns
+2.5 seconds after they spawn, we expect the number of enemies present in the
+scene to stabilize to 50. We can make sure about this by looking at the graph.
+
+To visualize the graph created from this custom performance monitor, run the
+project, switch to the editor while the project is running and open **Debugger >
+Monitors** at the bottom of the editor window. Scroll down to the newly
+available **Game** section and check **Enemies**. You should see a graph
+appearing as follows:
+
+.. :figure: img/custom_performance_monitors_graph_example.webp
+   :align: center
+   :alt: Example editor graph from a custom performance monitor
+
+   Example editor graph from a custom performance monitor
+
+The performance monitor is only queried if the project is being run from the editor.
+The second parameter is a :ref:`class_Callable`.
```

Comment:
This sentence seems out of place. I assume it's referring to `Performance.add_custom_monitor()`, but it's several paragraphs after that part of the code, I thought it was referring to something related to what you see in the debugger panel at first.